### PR TITLE
Core-107 click interactions and tracking events for KivaClassicBasicL…

### PR DESCRIPTION
**Pull request by BoulderBrains**
…oanCard

[Core-107](https://kiva.atlassian.net/browse/CORE-107)

- Click interaction added to borrower image by wrapping it in a router-link. 
      - There's a conditional in place that if allSharesReserved is true, we disable the < router-link > by converting it to a span tag.
- Tracking info added for borrower image click and "Read more" cta button click added based on spec in ticket.

### CODE REVIEW

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue [hunk 1]
```diff
@@ -9,30 +9,32 @@
...
 		/>
 
-		<!-- TODO: Once click interaction for this is in place, we need to disable the click
-		if allSharesReserved is true
-		ie.
-		:click="`${allSharesReserved ? 'correct redirect' : ''}`" -->
-		<borrower-image
-			v-if="!isLoading"
-			class="
+		<!-- Is there a way to display the click completely if allSharesAreReserved? -->
```
Comment (mcstover) [inline src/components/LoanCards/KivaClassicBasicLoanCard.vue]: You can use a v-if on the router-link and it's containing borrower-image then have the v-else on an duplicated instance of the image when you don't want to show a click...I do question why we would turn off the link at all in this scenario. I don't think we should be removing this link...

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue [hunk 2]
```diff
@@ -9,30 +9,33 @@
...
-			:images="[
-				{ width: 336, viewSize: 1024 },
-				{ width: 336, viewSize: 768 },
-				{ width: 416, viewSize: 480 },
-				{ width: 374, viewSize: 414 },
-				{ width: 335, viewSize: 375 },
-				{ width: 280 },
-			]"
-		/>
+		<!-- If allSharesReserved, disable link by make it a span -->
```
Comment (mcstover) [inline src/components/LoanCards/KivaClassicBasicLoanCard.vue:12]: Why do we need to remove this link at all?

Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:12]: The requested allSharesReserved state from Nathan is "disabled button, (no click targets) - and funding complete". 

I mentioned that it felt odd to block users from learning more about a borrower. But he wants to keep this block in place until we get the recommended loan cards on funded borrower pages.

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue [hunk 3]
```diff
@@ -3,52 +3,57 @@
...
-				:distribution-model="distributionModel"
-			/>
-			<summary-tag
-				v-if="countryName"
-				class="tw-absolute tw-bottom-2 tw-left-1"
-				:city="city"
-				:state="state"
-				:country-name="countryName"
+			<!-- If allSharesReserved, disable link by making it a span -->
+			<router-link
```
Comment (eddieferrer) [inline src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: Does this work? This is interesting syntax. I would expect this to be `<component :is="allSharesReserved ? 'span' : 'router-link'"`

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue [hunk 4]
```diff
@@ -3,52 +3,57 @@
...
+				:is="allSharesReserved ? 'span' : 'router-link'"
+				:to="`${!allSharesReserved ? `/lend/${loanId}` : diable}`"
```
Comment (eddieferrer) [inline src/components/LoanCards/KivaClassicBasicLoanCard.vue:18]: diable?

Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:18]: Thanks for calling out the redundancy, fixed that up.

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue (hunk 3 above)
Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: What I have in place does work, but I don't see a difference between what I have in there and what you suggested above. Am I misunderstanding you?

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue (hunk 2 above)
Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue]: Conversation discussed and resolved in Slack.

**Review by mcstover — APPROVED** [review]

**Comment by BoulderBrains** [comment]
I'd like to get this merged into master branch and then work on the additional changes requested in slack and captured in CORE-110 & CORE-111

The conclusion of that discussion was the following work. 
- We will add the “Learn more” text link after the loan use statement,
- When all loan shares are reserved, 
     - we will replace the CTA button with the “another lender has selected this loan” message
     -  Hide the “Learn more” text link from #1
     - Remove the link from the image _(This change is present in this code review)_

File: src/components/LoanCards/KivaClassicBasicLoanCard.vue (hunk 3 above)
Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: This is rendering in storybook. 
**When router-link renders as a link (while loan is fundraising)**
Screen Shot 2021-09-24 at 1 54 31 PM

**When router-link renders as a span (while allSharesReserved)**
Screen Shot 2021-09-24 at 1 55 18 PM

Comment (mcstover) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: Is the `span` still clickable? Could you also empty the 'to' field in this case?

Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: I'll be sure to verify this functionality in dev across the major browsers.

Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: <router-link is used in the same way within [CheckoutItemImg.vue on lines 2 & 3](https://github.com/kiva/ui/blob/master/src/components/Checkout/CheckoutItemImg.vue#L3). That change was introduced about 8 months ago.

Comment (BoulderBrains) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: @mcstover I had that :to field return an empty string at an earlier point in my code review, but it was raised that it was a little redundant to have the allSharesReserved check on both fields, so I removed that. The Span is not clickable when returned and the mouse over event is that of a span. Is it prefered to clear out to :to field in the allSharesReserved state?

Comment (mcstover) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: SOunds good, thanks for checking! It's fine as is.

Comment (eddieferrer) [reply src/components/LoanCards/KivaClassicBasicLoanCard.vue:16]: > What I have in place does work, but I don't see a difference between what I have in there and what you suggested above. Am I misunderstanding you?
> 
> `<router-link :is="allSharesReserved ? 'span' : 'router-link'"`

@BoulderBrains I meant to literally make it: 
`<component
:is="allSharesReserved ? 'span' : 'router-link'"
:to="`/lend/${loanId}`"
v-kv-track-event="['Lending', 'click-Read more', 'Photo', loanId]"
>`

But looks like you tested it and it's working, so it's fine as is. :thumbs

**Pull request merged**